File: tests/fixtures/generic/clean/taint_ldap_clean.py
# @audit-fixture
# @rule: taint_ldap
# @category: security
# @expected: clean
from flask import request
import ldap
@app.route("/search")
def search():
    user = ldap.filter.escape_filter_chars(request.args.get("user"))
    conn.search_s("dc=example", ldap.SCOPE_SUBTREE, f"(uid={user})")

File: tests/fixtures/generic/clean/taint_ldap_clean.js
// @audit-fixture
// @rule: taint_ldap
// @category: security
// @expected: clean
const ldap = require("ldapjs");
const express = require("express");
const app = express();
app.post("/auth", (req, res) => {
    const user = ldap.escapeFilter(req.body.user);
    const client = ldap.createClient({ url: "ldap://localhost" });
    client.search("dc=example,dc=com", { filter: "(uid=" + user + ")" }, (err, r) => {
        res.json({ ok: !err });
    });
});
